eventing/samples/github-source/README.md

```diff
@@ -0,0 +1,160 @@
+  - Ensure Knative Serving is [configured with a domain
+    name](https://github.com/knative/docs/blob/master/serving/using-a-custom-domain.md)
+    that allows GitHub to call into the cluster.
+  - If you're using GKE, you'll also want to [assign a static IP address](https://github.com/knative/docs/blob/master/serving/gke-assigning-static-ip-address.md).
+1. Install [Knative
+  Eventing](https://github.com/knative/docs/tree/master/eventing).
+1. Create a `Channel`. You can use your own `Channel` or use the
+   provided sample, which creates a channel called `githubchannel`. If
+   you use your own `Channel` with a different name, then you will
+   need to alter other commands later.
+
+```shell
+kubectl -n default apply -f eventing/samples/github-source/channel.yaml
+```
+
+### Deploy Event Sources
+
+1. Deploy the `GitHubSource` controller as part of eventing-source's
+   controller. This makes GitHub events available for subscriptions.
+
+```shell
+kubectl apply -f https://knative-releases.storage.googleapis.com/eventing-sources/latest/release.yaml
+```
+
+### Create GitHub Tokens
+
+1.  Create a [personal access
+    token](https://github.com/settings/tokens) for GitHub that the
+    GitHub source can use to register webhooks with the GitHub
+    API. Also decide on a secret token that your code will use to
+    authenticate the incoming webhooks from GitHub (*secretToken*).
+    
+    The token can be named anything you find convenient. The Source
+    requires `repo:public_repo` and `admin:repo_hook`, to let it fire
+    events from your public repositories and to create webhooks for
+    those repositories. Copy and save this token; GitHub will force
+    you to generate it again if misplaced.
+
+    Here's an example for a token named "GitHubSource Sample" with the
+    the recommended scopes:
+
+    ![GitHub UI](personal_access_token.png "GitHub personal access token screenshot")
+
+    Update `githubsecret.yaml` with those
+    values. If  your generated access token is `'asdfasfdsaf'` and you choose
+    your *secretToken* as `'personal_access_token_value'`, you'd modify
+    `githubsecret.yaml` like so:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: githubsecret
+type: Opaque
+stringData:
+  accessToken: asdfasfdsaf
+  secretToken: personal_access_token_value
+```
+
+Hint: you can makeup a random *accessToken* with:
+    
+```shell
+head -c 8 /dev/urandom | base64
+```
+
+Then, apply the githubsecret using `kubectl`:
+    
+```shell
+kubectl -n default apply -f eventing/samples/github-source/githubsecret.yaml
+```
+
+### Create Event Source for GitHub Events
+
+1. In order to receive GitHub events, you have to create a concrete
+   Event Source for a specific namespace. Be sure to replace the
+   `ownerAndRepository` value with a valid GitHub public repository
+   owned by your GitHub user. If you are using a different `Secret`
+   name, `Channel`, or `Service Account`, modify the yaml accordingly.
```

Comment:
In a number of the examples in serving, we include the relevant YAML in-line so that it's easier for the user to see. I don't know if it's worth including the 21 lines here, but they seem useful enough that I might recommend it. For an example, see https://github.com/knative/docs/tree/master/serving/samples/build-private-repo-go#creating-a-deploy-key